parlai/tasks/genderation_bias/agents.py

```diff
@@ -0,0 +1,327 @@
+# `--task internal:gender_multiclass:controllable_task:<ORIGINAL TASK NAME>`.
+`--task genderation_bias:controllable_task:<ORIGINAL TASK NAME>`.
+
+As an example, try running:
+
+`parlai display_data -t genderation_bias:controllable_task:convai2`
+"""
+
+from parlai.core.message import Message
+from parlai.core.opt import Opt
+from parlai.core.teachers import FixedDialogTeacher
+from parlai.utils.io import PathManager
+import parlai.utils.logging as logging
+from parlai.utils.typing import TShared
+
+from parlai.tasks.genderation_bias.build import build
+from parlai.tasks.genderation_bias.utils import (
+    flatten_and_classify,
+    get_original_task_module,
+)
+
+from copy import deepcopy
+import datetime
+import glob
+import json
+import os
+from tqdm import tqdm
+from typing import List, Optional, Tuple
+
+
+class ControllableTaskTeacher(FixedDialogTeacher):
+    """
+    Generates a controllable_gen version of a ParlAI task, i.e., for tasks with multi-
+    turn dialogues (episodes), this will generate a task with single example episodes,
+    in which we append to the context a special classification token.
+    """
+
+    @staticmethod
+    def add_cmdline_args(parser):
+        flattened = parser.add_argument_group('Flattening Args')
+        flattened.add_argument(
+            '--flatten-include-labels',
+            type='bool',
+            default=True,
+            help='Include labels in the history when flattening an episode',
+        )
+        flattened.add_argument(
+            '--flatten-delimiter',
+            type=str,
+            default='\n',
+            help='How to join the dialogue history from previous turns.',
+        )
+        flattened.add_argument(
+            '--flatten-max-context-length',
+            type=int,
+            default=-1,
+            help='Maximum number of utterances to include per episode. '
+            'Default -1 keeps all.',
+        )
+        agent = parser.add_argument_group('Controllable Gen Teacher Args')
+        agent.add_argument(
+            '--invalidate-cache',
+            type='bool',
+            default=False,
+            help='Set this to True to rebuild the data (may want to do this if '
+            'original data has changed or you want to rebuild with new options)',
+        )
+        agent.add_argument(
+            '--max-examples',
+            type=int,
+            default=-1,
+            help='If greater than zero, will stop building after a certain num of exs',
+        )
+        agent.add_argument(
+            '--four-class',
+            type='bool',
+            default=True,
+            help='Use 4 class classification for gendered word list (f0m0/f0m1/f1m0/f1m1). '
+            'Set false to use 3-class classification (MALE/FEMALE/NEUTRAL)',
+        )
+        agent.add_argument(
+            '--fixed-control',
+            type=str,
+            default='',
+            help='Always append this fixed control variable, good for deploy time.',
```

Comment:
hmm "variable" seems strange in this context (maybe because it's fixed, not variable, ha) - maybe "string"? Will leave this up to you